ocs_ci/ocs/quay_operator.py

```diff
@@ -0,0 +1,132 @@
+import logging
+from time import sleep
+
+from ocs_ci.helpers import helpers
+from ocs_ci.helpers.helpers import storagecluster_independent_check
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.resources.ocs import OCS
+from ocs_ci.ocs.utils import get_pod_name_by_pattern
+from ocs_ci.utility import templating
+from ocs_ci.utility.utils import TimeoutSampler, exec_cmd, run_cmd
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.exceptions import TimeoutExpiredError
+
+logger = logging.getLogger(__name__)
+
+
+class QuayOperator(object):
+    """
+    Quay operator class
+
+    """
+
+    def __init__(self):
+        """
+        Initializer function
+
+        """
+        self.namespace = "openshift-operators"
+        self.quay_operator = None
+        self.quay_registry = None
+        self.quay_pod_obj = OCP(kind="pod", namespace=self.namespace)
+        self.quay_registry_name = ""
+        self.quay_operator_csv = ""
+        self.sc_default = False
+        self.sc_name = (
+            constants.DEFAULT_EXTERNAL_MODE_STORAGECLASS_RBD
+            if storagecluster_independent_check()
+            else constants.DEFAULT_STORAGECLASS_RBD
+        )
+
+    def setup_quay_operator(self):
+        """
+        Deploys Quay operator
+
+        """
+        quay_operator_data = templating.load_yaml(file=constants.QUAY_SUB)
+        self.quay_operator_csv = quay_operator_data["spec"]["startingCSV"]
+        self.quay_operator = OCS(**quay_operator_data)
+        logger.info(f"Installing Quay operator: {self.quay_operator.name}")
+        self.quay_operator.create()
+        for quay_pod in TimeoutSampler(
+            300, 10, get_pod_name_by_pattern, "quay-operator", self.namespace
+        ):
+            if quay_pod:
+                assert self.quay_pod_obj.wait_for_resource(
```

Comment:
wait for resource will throws exception after the timeout, do we still need assert here?

Reply:
removed assert.